src/App.tsx

```diff
@@ -8,7 +13,81 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [selectedTodoId, setSelectedTodoId] = useState<number | null>(null);
+  const [query, setQuery] = useState('');
+  const [filterStatus, setFilterStatus] = useState('all');
+  const [debouncedQuery, setDebouncedQuery] = useState('');
+
+  useEffect(() => {
+    const loadTodos = async () => {
+      try {
+        const loadedTodos = await getTodos();
+
+        setTodos(loadedTodos);
+      } catch (error) {
+        setTodos([]);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    loadTodos();
+  }, []);
+
+  const debounce = (
+    func: React.Dispatch<React.SetStateAction<string>>,
+    delay: number,
+  ) => {
+    let timerId = 0;
+
+    return (...args: unknown[]) => {
+      if (timerId) {
+        clearTimeout(timerId);
+      }
+
+      timerId = setTimeout(func, delay, ...args);
+    };
+  };
```

Comment:
You should move this function to `helpers/debounce.ts` file. So you will not create this function on each `App` re-render